src/libServer/ScillaIPCServer.cpp

```diff
@@ -15,64 +15,78 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-#include "ScillaIPCServer.h"
+#include <jsonrpccpp/server/connectors/unixdomainsocketserver.h>
+
 #include "libPersistence/ContractStorage2.h"
 #include "libUtils/DataConversion.h"
 
-using namespace jsonrpc;
+#include "ScillaIPCServer.h"
+
 using namespace std;
 using namespace Contract;
 
-ScillaIPCServer::ScillaIPCServer(UnixDomainSocketServer &server,
-                                 const dev::h160 &contract_address)
-    : AbstractServer<ScillaIPCServer>(server, JSONRPC_SERVER_V2) {
-  this->bindAndAddMethod(Procedure("fetchStateValue", PARAMS_BY_NAME,
-                                   JSON_STRING, "query", JSON_STRING, NULL),
-                         &ScillaIPCServer::fetchStateValueI);
-
-  this->bindAndAddMethod(
-      Procedure("updateStateValue", PARAMS_BY_NAME, JSON_BOOLEAN, "query",
-                JSON_STRING, "value", JSON_STRING, NULL),
+ScillaIPCServer::ScillaIPCServer(const dev::h160 &contrAddr,
+                                 jsonrpc::UnixDomainSocketServer &conn,
+                                 jsonrpc::serverVersion_t type)
+    : jsonrpc::AbstractServer<ScillaIPCServer>(conn, type),
+      m_contrAddr(contrAddr) {
+  // These signatures matche that of the actual functions below.
+  bindAndAddMethod(
+      jsonrpc::Procedure("fetchStateValue", jsonrpc::PARAMS_BY_NAME,
+                         jsonrpc::JSON_OBJECT, "query", jsonrpc::JSON_STRING,
+                         NULL),
+      &ScillaIPCServer::fetchStateValueI);
+  bindAndAddMethod(
+      jsonrpc::Procedure("updateStateValue", jsonrpc::PARAMS_BY_NAME,
+                         jsonrpc::JSON_STRING, "query", jsonrpc::JSON_STRING,
+                         "value", jsonrpc::JSON_STRING, NULL),
       &ScillaIPCServer::updateStateValueI);
+}
 
-  this->bindAndAddMethod(Procedure("testServerRPC", PARAMS_BY_NAME, JSON_STRING,
-                                   "query", JSON_STRING, NULL),
-                         &ScillaIPCServer::testServerRPCI);
+void ScillaIPCServer::fetchStateValueI(const Json::Value &request,
+                                       Json::Value &response) {
+  std::string value;
+  bool found;
+  if (!fetchStateValue(request["query"].asString(), value, found)) {
+    throw jsonrpc::JsonRpcException("Fetching state value failed");
+  }
 
-  this->contract_address = contract_address;
+  // Prepare the result and finish.
+  response.clear();
+  response.append(Json::Value(found));
+  response.append(Json::Value(value));
 }
 
-void ScillaIPCServer::setContractAddress(dev::h160 &address) {
```

Comment:
You might want to check with @kaikawaliu whether it is ok to remove this method, because it was after discussing with him that I had put it there.

Reply:
`m_contrAddr` is a `const` private member. I can't imagine what advantage having an accessor for it would provide. If @kaikawaliu insists, I'll add it.

Reply:
Not an accessor, the ability to set it.

Reply:
Would there be a situation where it needs to be set _after_ the server object is instantiated?

Reply:
That's what needs to be checked with @kaikawaliu

Reply:
Yes that's needed, as long as the scilla query doesn't include the address itself and we need the address to compose the index (you can see the param is needed to be fed to the ContractStorage2 functions), we have to somewhere provide the address to the ipc somewhere.

Reply:
the addr needed to be change frequently, so the set function for the address is needed here

Reply:
Ah that makes sense. I thought once the server is initialized, it will be for the same address. I'll make the change. Thank you.

Reply:
@kaikawaliu this is now done.